### PR TITLE
Feature/call to api launch details

### PR DESCRIPTION
**Pull request by vovkis13**
[Implement call to API for detailed view of launches](https://trello.com/c/Svv6lOE7/8-implement-call-to-api-for-detailed-view-of-launches)

added selector getActiveSlideData for selection details of active slide from redux
added call to redux store from LaunchDetailsPage component
added props to LaunchDetailsPage child components with launch details
added route `/details` and  changed prop `to` of SliderCardFooter component
deleted `/mocks` from LaunchDetailsPage component

in selectors.ts type `any` will be changed to LaunchDetails and StoreState further

### CODE REVIEW

File: src/components/LaunchDetailsPage/selectors/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { LaunchDetails } from '../types/LaunchDetails';
+
+export const getActiveSlideData: any = createSelector(
```
Comment (vovkis13) [inline src/components/LaunchDetailsPage/selectors/selectors.ts:4]: `any` will be changed to  `LaunchDetails`

File: src/components/LaunchDetailsPage/selectors/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  [
+    (state: any) => state.slider.slides,
+    (state: any) => state.slider.activeSlideId,
```
Comment (vovkis13) [inline src/components/LaunchDetailsPage/selectors/selectors.ts:7]: `any` will be changed to `StateStore` after configuring redux

File: src/components/LaunchDetailsPage/selectors/selectors.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { createSelector } from '@reduxjs/toolkit';
```
Comment (topsart) [inline src/components/LaunchDetailsPage/selectors/selectors.ts:1]: Do we use createSelector from redux-toolkit?

Comment (vovkis13) [reply src/components/LaunchDetailsPage/selectors/selectors.ts:1]: @topsart, is it ok? I remember you approved using of  createSelector from Reselect library which reexports by @reduxjs/toolkit

File: src/components/LaunchDetailsPage/LaunchDetailsPage.tsx [hunk 4]
```diff
@@ -8,10 +9,13 @@ import {
 } from '@mui/material';
 import { SecondaryButton } from '../../shared/components/SecondaryButton';
 import { LaunchCardContent } from './components/LaunchCardContent';
-import { launchDetails } from './mocks/launchDetails';
+import { getActiveSlideData } from './selectors/selectors';
+import type { LaunchDetails } from './types/LaunchDetails';
 import './LaunchDetailsPage.scss';
 
 export const LaunchDetailsPage = () => {
+  const details: LaunchDetails = useSelector(getActiveSlideData);
```
Comment (PietyE) [inline src/components/LaunchDetailsPage/LaunchDetailsPage.tsx:17]: It is depend on code style but we can use destructuring here

File: src/components/LaunchDetailsPage/LaunchDetailsPage.tsx [hunk 5]
```diff
@@ -20,14 +24,14 @@ export const LaunchDetailsPage = () => {
           <Grid container columns={{ xs: 4, sm: 8, md: 12 }}>
             <Grid item xs={4} sm={3} md={4}>
               <CardMedia
-                src={launchDetails.image}
-                alt={launchDetails.name}
+                src={details.image}
+                alt={details.name}
```
Comment (PietyE) [inline src/components/LaunchDetailsPage/LaunchDetailsPage.tsx:28]: so here we wont repeat details and then his field

File: src/components/LaunchDetailsPage/LaunchDetailsPage.tsx (hunk 4 above)
Comment (vovkis13) [reply src/components/LaunchDetailsPage/LaunchDetailsPage.tsx:17]: I need the whole object to pass it into the prop in line 34

**Comment by vovkis13** [comment]
> PR looks good. Just reset last commit with deleting mocks

File: src/components/LaunchDetailsPage/LaunchDetailsPage.scss [hunk 6]
```diff
@@ -1,3 +1,11 @@
+.launch-details-card {
+  margin-top: 1rem;
+}
+
+.launch-details-grid {
+  min-height: 60vh;
```
Comment (kerrimov) [inline src/components/LaunchDetailsPage/LaunchDetailsPage.scss:6]: Why vh is used here instead rem?

Comment (vovkis13) [reply src/components/LaunchDetailsPage/LaunchDetailsPage.scss:6]: I try to make the card fill  at least 60% of screen without depending on screen resolution

**Review by kerrimov — APPROVED** [review]

**Pull request merged**